src/community/wmts-multi-dimensional/src/main/java/org/geoserver/gwc/wmts/MultiDimensionalExtension.java

```diff
@@ -198,8 +217,48 @@ private Domains getDomains(SimpleConveyor conveyor) throws Exception {
         if (requestedDomains == ALL_DOMAINS || requestedDomains.contains(SPACE_DIMENSION)) {
             spatialDomain = DimensionsUtils.getBounds(resource, filter);  
         } 
+        // get the threshold
+        int expandLimit = getExpandlimit(resource, conveyor.getParameter(EXPAND_LIMIT, false));
         // encode the domains
-        return new Domains(dimensions, layerInfo, spatialDomain, SimplifyingFilterVisitor.simplify(filter));
+        return new Domains(dimensions, layerInfo, spatialDomain, SimplifyingFilterVisitor.simplify(filter), expandLimit);
+    }
+
+    private int getExpandlimit(ResourceInfo resource, Object clientExpandLimit) throws OWSException {
+        WMTSInfo wmts = geoServer.getService(WMTSInfo.class);
+        int expandLimitMax = getConfigredExpansionLimit(resource, wmts, EXPAND_LIMIT_MAX_KEY, EXPAND_LIMIT_MAX_DEFAULT);
+        int expandLimitDefault = getConfigredExpansionLimit(resource, wmts, EXPAND_LIMIT_KEY, EXPAND_LIMIT_DEFAULT);
+
+        if (clientExpandLimit != null) {
+            Integer value = Converters.convert(clientExpandLimit, Integer.class);
+            if (value == null) {
+                throw new OWSException(400, "InvalidParameterValue", EXPAND_LIMIT, "Invalid " + EXPAND_LIMIT + " " +
+                        "value " + clientExpandLimit + ", expected an integer value");
+            }
+            if (value < expandLimitMax) {
+                return value;
+            } else {
+                LOGGER.log(Level.INFO, "Client provided a expand value higher than the configured max, using the " +
```

Comment:
The spec will need to be explicit about this behavior, since there is now way for the client to know about the max value defined in the server this may cause some misunderstanding. Bu tI'm not sure about the practical consequences of that.

Reply:
![image](https://user-images.githubusercontent.com/325433/37614777-e8c463e2-2bab-11e8-8daf-9ff0df6dbfa1.png)